examples/text-generation/textgen_pipeline/pipeline.py

```diff
@@ -0,0 +1,56 @@
+import os
+import sys
+
+import torch
+from transformers import TextGenerationPipeline
+
+
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), os.path.pardir)))
+try:
+    from utils import initialize_model
+except:
+    raise
+
+
+class GaudiTextGenerationPipeline(TextGenerationPipeline):
+    def __init__(self, args, logger):
+        self.model, self.tokenizer, self.generation_config = initialize_model(args, logger)
+
+        self.device = args.device
+
+        if args.do_sample:
+            self.generation_config.temperature = args.temperature
+            self.generation_config.top_p = args.top_p
```

Comment:
`generation_config` is this structure:

https://github.com/huggingface/optimum-habana/blob/8726c1ef4181b3004ced8aad776e0f3b0928d304/optimum/habana/transformers/generation/configuration_utils.py

which is filled here:
https://github.com/huggingface/optimum-habana/blob/8726c1ef4181b3004ced8aad776e0f3b0928d304/examples/text-generation/utils.py#L316


`temperature` and `top_p` dont seem to be part of `generation_config`, and where are they used? I cannot find them in this file:
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/generation/utils.py

Reply:
`temperature` and `top_p` are attributes of the parent class of https://github.com/huggingface/optimum-habana/blob/8726c1ef4181b3004ced8aad776e0f3b0928d304/optimum/habana/transformers/generation/configuration_utils.py.
Yes, they are not part of https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/generation/utils.py but the quality of the generated output changes with these parameters.

Reply:
Ah makes sense.. parent class. Sounds good